### ISSUE TITLE
predict method not allowed on 1.102

### ISSUE THREAD

**Issue by jessegoodier** [kubecost/kubectl-cost#153]
Not sure if this is user error or not, but:

```sh
k cost predict -f cpu-limit-eq-request.yaml --predict-speccost-path https://demo.kubecost.xyz/model/prediction/speccost
```

yields: 

```log
Error: Failed querying the speccost API. This API requires a version of Kubecost >= 1.101, which may be why this query failed. If running Kubecost v1.100, you can downgrade kubectl cost to v0.4 for old-style prediction. Error: failed to port forward query: received non-200 status code 405 and data: <html>
<head><title>405 Not Allowed</title></head>
<body>
<center><h1>405 Not Allowed</h1></center>
<hr><center>nginx/1.22.1</center>
</body>
</html>
```

**Comment by dwbrown2** [comment]
@jessegoodier do you know what version of kubectl cost you're running?

@michaelmdresser @teevans for triage

**Comment by jessegoodier** [comment]
Update, this is not the way to use predict. Though there still seems to be an issue with predic. Is it my client?

this is correct, requires a proper context:

```
k cost predict -f memory-limit-eq-request.yaml
```

using kubectl v1.25.4

error:

```
k cost predict -f memory-limit-eq-request.yaml           
Error: Failed querying the speccost API. This API requires a version of Kubecost >= 1.101, which may be why this query failed. If running Kubecost v1.100, you can downgrade kubectl cost to v0.4 for old-style prediction. Error: failed to port forward query: received non-200 status code 500 and data: Failed to predict cost impact of specs: cannot predict impact of nil spec
```

**Comment by kc-creynold** [comment]
some more info on what we are seeing:

```
kubectl config set-context --current --namespace=kubecost
Context "arn:aws:eks:us-west-2:297945954695:cluster/kc-demo-prod" modified.
$ chris@Christophers-MacBook-Pro  ~/src/se-demo/predict_samples   predict_samples_initial  kubectl cost predict -f small.yaml --log-level debug
2023-04-17T13:12:49.154625-04:00 DBG ServiceName set to: kubecost-cost-analyzer
2023-04-17T13:12:49.154746-04:00 DBG KubecostNamespace set to: kubecost
2023-04-17T13:12:50.177065-04:00 DBG checking readiness of 'kubecost-cost-analyzer-74d4f67bdf-qb2km'
2023-04-17T13:12:50.177091-04:00 DBG selected pod to forward: kubecost-cost-analyzer-74d4f67bdf-qb2km
2023-04-17T13:12:50.426228-04:00 DBG Port-forward set up at: http://localhost:58222/
2023-04-17T13:12:50.432428-04:00 DBG Executing GET to: http://localhost:58222/model/clusterInfo
2023-04-17T13:12:50.678468-04:00 DBG Cluster ID for query set to: kc-demo-prod
2023-04-17T13:12:50.678526-04:00 DBG Executing POST to: http://localhost:58222/model/prediction/speccost?clusterID=kc-demo-prod&defaultNamespace=kubecost&noUsage=false&windowAvgUsage=2d&windowResourceCost=7d+offset+48h
Error: Failed querying the speccost API. This API requires a version of Kubecost >= 1.101, which may be why this query failed. If running Kubecost v1.100, you can downgrade kubectl cost to v0.4 for old-style prediction. Error: failed to port forward query: received non-200 status code 500 and data: Failed to predict cost impact of specs: cannot predict impact o
```f nil spec

**Comment by teevans** [comment]
@michaelmdresser  - Can you take a look here?

**Comment by michaelmdresser** [comment]
@kc-creynold @jessegoodier is this broken in nightly? We have a PR slated for release in 1.103 that should fix this: https://github.com/kubecost/kubecost-cost-model/pull/1383

**Comment by jessegoodier** [comment]
quick test works in nightly.

**Comment by michaelmdresser** [comment]
Closing as this should be fixed in v1.103 of Kubecost. If an error is still occurring, please reopen this issue.
